zuul-core/src/main/java/com/netflix/zuul/message/Headers.java

```diff
@@ -499,4 +499,20 @@ private void truncate(int i) {
             values.remove(k);
         }
     }
+
+    private String sanitizeField(String value) {
```

Comment:
sanitization is an attack vector.   I feel that if the value is bad, the header should be thrown away.  (or the respond with a 400).

Reply:
I'm open to throwing an exception in that method instead of sanitizing. Curious what the attack vector is here?

Reply:
Mutating the response I think it incorrect.   What if by stripping chars, we end up making something that would previously be blocked?   We need to only produce good data.   I think throwing an exception is reasonable.

Also, if we get data in from netty, it may be "invalid", so this would only protect stuff produced by our filters.  I think the validation would need to happen down at the netty layer.

Reply:
Yeah, I think you're right. I'll make it throw an exception.

On the second point, I believe the Netty HTTP codec will protect us from incoming parsing errors but this will protect us from the filters settings bad headers.